src/ResourceManager/Network/Commands.Network/DdosProtectionPlan/GetAzureRMDdosProtectionPlanCommand.cs

```diff
@@ -0,0 +1,88 @@
+// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using AutoMapper;
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Rest.Azure;
+using Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Management.Network;
+using Microsoft.Azure.Management.Network.Models;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using CNM = Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.Network.Automation
+{
+    [Cmdlet(VerbsCommon.Get, "AzureRmDdosProtectionPlan"), OutputType(typeof(PSDdosProtectionPlan))]
+    public partial class GetAzureRmDdosProtectionPlan : NetworkBaseCmdlet
```

Comment:
@murilogr is this a tracked resource? If so, we should add an additional parameter set that allows the user to pass just `ResourceId`, and you can use common code to strip out the `ResourceGroupName` and `Name`. This allows users to call `Get-AzureRmResource` and pipe the result to this cmdlet.

Reply:
What do you mean by tracked resource?

Reply:
@murilogr is this a resource that has a `ResourceId` that the user can retrieve from a generic REST call? (_i.e._, using `Get-AzureRmResource` -- [check out this doc](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/piping-in-powershell.md#using-the-resourceid-parameter))